src/cross_section.F90

```diff
@@ -84,21 +88,22 @@ subroutine calculate_xs(p)
           end if
         end if
 
-        ! ========================================================================
+        ! ======================================================================
         ! CALCULATE MICROSCOPIC CROSS SECTION
 
         ! Determine microscopic cross sections for this nuclide
         i_nuclide = mat % nuclide(i)
 
         ! Calculate microscopic cross section for this nuclide
         if (p % E /= micro_xs(i_nuclide) % last_E &
-             .or. p % sqrtkT /= micro_xs(i_nuclide) % last_sqrtkT) then
-          call calculate_nuclide_xs(i_nuclide, i_sab, p % E, i_grid, p % sqrtkT)
-        else if (i_sab /= micro_xs(i_nuclide) % last_index_sab) then
-          call calculate_nuclide_xs(i_nuclide, i_sab, p % E, i_grid, p % sqrtkT)
+             .or. p % sqrtkT /= micro_xs(i_nuclide) % last_sqrtkT &
+             .or. i_sab /= micro_xs(i_nuclide) % index_sab &
+             .or. sab_frac /= micro_xs(i_nuclide) % sab_frac) then
```

Comment:
I was actually talking about the local variable `sab_frac` in `calculate_xs` which can also be undefined.